### PR TITLE
Refactor the version control

### PR DESCRIPTION
**Pull request by yinan-symphony**
Separate the two swadl management process, disk and management api. Use version control when manage swadl through management api only. This simplifies the complexity of the version control and the maintenance in the future.

Version is controlled on server side and becomes mandatory in DB, currently the version value is a timestamp at macro precision.

Reorganise the two datasource configuration in application.yml.

The DB used for version control is not required when the management api is disabled (management token is not set).

### Description
Please put here the intent of your pull request.

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [ ] Referenced a ticket in the PR title or description
- [ ] Filled properly the description and dependencies, if any
- [ ] Unit/Integration tests updated or added
- [ ] Javadoc added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/ConditionalOnPropertyNotEmpty.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+      Map<String, Object> attrs = metadata.getAnnotationAttributes(ConditionalOnPropertyNotEmpty.class.getName());
+      if (attrs == null) {
+        return false;
+      }
+      String propertyName = (String) attrs.get("value");
+      String val = context.getEnvironment().getProperty(propertyName);
+      return val != null && !val.trim().isEmpty();
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/ConditionalOnPropertyNotEmpty.java:30]: can't you directly use StringUtils.isNotBlank() / .isNotEmpty()?

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDeployer.java [hunk 2]
```diff
@@ -60,119 +51,41 @@ public void addAllWorkflowsFromFolder(Path path) {
...
-        }
-
-      } else {
-        log.debug("Unknown event: {}", event);
-      }
-    }
-  }
-
-  public void addWorkflow(Path workflowFile, WorkflowMgtAction action) throws IOException, ProcessingException {
+  void addWorkflow(Path workflowFile) throws IOException, ProcessingException {
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDeployer.java:54]: can be private

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/converter/BiConverter.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+public interface BiConverter<S, K, T> extends BiFunction<S, K, T> {
+  /**
+   * The source type for which this is going to convert.
+   *
+   * @return the source class type
+   */
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/converter/BiConverter.java:14]: is it safe to use hardcoded indexes? 0 and 2 ?

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/exception/GlobalExceptionHandler.java [hunk 4]
```diff
@@ -21,7 +26,7 @@ public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
   public ResponseEntity<ErrorResponse> handle(Throwable exception) {
     log.error("Internal server error: [{}]", exception.getMessage());
     log.debug("", exception);
-    return handle(exception.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+    return handle("Internal server error, something goes wrong.", HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (symphony-soufiane) [inline workflow-bot-app/src/main/java/com/symphony/bdk/workflow/exception/GlobalExceptionHandler.java:29]:
```suggestion
    return handle("Internal server error, something went wrong.", HttpStatus.INTERNAL_SERVER_ERROR);
```

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDeployer.java (hunk 2 above)
Comment (yinan-symphony) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDeployer.java:54]: it is used in a test, so leave it in package level

File: workflow-bot-app/src/main/java/com/symphony/bdk/workflow/converter/BiConverter.java (hunk 3 above)
Comment (yinan-symphony) [reply workflow-bot-app/src/main/java/com/symphony/bdk/workflow/converter/BiConverter.java:14]: yes, safe

**Review by symphony-soufiane — APPROVED** [review]

**Pull request merged**